modules/mesh-task/main.tf

```diff
@@ -69,7 +69,7 @@ locals {
   ]
 
   defaulted_check_containers = length(var.checks) == 0 ? [for def in local.container_defs_with_depends_on : def.name
-  if contains(keys(def), "essential") && contains(keys(def), "healthCheck")] : []
+  if contains(keys(def), "essential") && contains(keys(def), "healthCheck") && try(def.healthCheck, null)] : []
```

Comment:
This change looks good to me.

I think that we need to add a new test to the [basic acceptance tests](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/test/acceptance/tests/basic/basic_test.go) to ensure that no `health-sync` container is injected for `container_definitions` where `healthCheck = null`.

This will help us to catch future regressions in this behaviour. I'm happy to help contribute to the testing if this is too much of an ask.

Reply:
Agree that having a test would be benefic, it's not too much to ask of course :) Unfortunately my scarse knowledge of Go and limited availability in the next weeks won't allow me to implement it in a time manner, so if/when you can I really apreciate the contribution with the test in this PR. I'll try for the next ones to make room to refresh my GO knowledge and include proper tests.

Reply:
No worries. I will add some acceptance tests to this PR. I probably won't get to it unitl the new year though due to limited availability over the holdiays.

Hopefully you are unblocked on this for now.